File: app/src/main/java/com/github/pavlospt/microsoftassignment/di/components/AssignmentRetrofitComponent.java
package com.github.pavlospt.microsoftassignment.di.components;

import com.github.pavlospt.microsoftassignment.api.service.BeersService;
import com.github.pavlospt.microsoftassignment.di.modules.ApiModule;
import com.github.pavlospt.microsoftassignment.di.modules.AssignmentDAOModule;
import dagger.Component;
import javax.inject.Singleton;

@Singleton
@Component(modules = {
    AssignmentDAOModule.class, ApiModule.class
})
public interface AssignmentRetrofitComponent {
  BeersService beersService();
}

File: app/src/main/java/com/github/pavlospt/microsoftassignment/di/components/AssignmentRetrofitComponent.kt
package com.github.pavlospt.microsoftassignment.di.components

import com.github.pavlospt.microsoftassignment.repo.remote.services.BeersService
import com.github.pavlospt.microsoftassignment.di.modules.ApiModule
import com.github.pavlospt.microsoftassignment.di.modules.AssignmentRetrofitModule
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AssignmentRetrofitModule::class, ApiModule::class))
interface AssignmentRetrofitComponent {
  fun beersService(): BeersService
}
